Src/FluentAssertions.Json/ObjectAssertionsExtensions.cs

```diff
@@ -0,0 +1,95 @@
+        /// is needed. If the phrase does not start with the word <i>because</i>, it is prepended automatically.
+        /// </param>
+        /// <param name="becauseArgs">
+        /// Zero or more objects to format using the placeholders in <see cref="because" />.
+        /// </param>
+        public static AndConstraint<ObjectAssertions> BeJsonSerializable(this ObjectAssertions assertions, string because = "", params object[] becauseArgs)
+        {
+            return BeJsonSerializable<object>(assertions, options => options, because, becauseArgs);
+        }
+
+        /// <summary>
+        /// Asserts that an object can be serialized and deserialized using the JSON serializer and that it stills retains
+        /// the values of all members.
+        /// </summary>
+        /// <param name="assertions"></param>
+        /// <param name="because">
+        /// A formatted phrase as is supported by <see cref="string.Format(string,object[])" /> explaining why the assertion 
+        /// is needed. If the phrase does not start with the word <i>because</i>, it is prepended automatically.
+        /// </param>
+        /// <param name="becauseArgs">
+        /// Zero or more objects to format using the placeholders in <see cref="because" />.
+        /// </param>
+        public static AndConstraint<ObjectAssertions> BeJsonSerializable<T>(this ObjectAssertions assertions, string because = "", params object[] becauseArgs)
+        {
+            return BeJsonSerializable<T>(assertions, options => options, because, becauseArgs);
+        }
+
+        /// <summary>
+        /// Asserts that an object can be serialized and deserialized using the JSON serializer and that it stills retains
+        /// the values of all members.
+        /// </summary>
+        /// <param name="options"></param>
+        /// <param name="because">
+        /// A formatted phrase as is supported by <see cref="string.Format(string,object[])" /> explaining why the assertion 
+        /// is needed. If the phrase does not start with the word <i>because</i>, it is prepended automatically.
+        /// </param>
+        /// <param name="becauseArgs">
+        /// Zero or more objects to format using the placeholders in <see cref="because" />.
+        /// </param>
+        /// <param name="assertions"></param>
+        public static AndConstraint<ObjectAssertions> BeJsonSerializable<T>(this ObjectAssertions assertions, Func<EquivalencyAssertionOptions<T>, EquivalencyAssertionOptions<T>> options, string because = "", params object[] becauseArgs)
+        {
+            try
+            {
+                var deserializedObject = CreateCloneUsingJsonSerializer(assertions.Subject);
+
+                var defaultOptions = AssertionOptions.CloneDefaults<T>()
+                    .RespectingRuntimeTypes()
+                    .IncludingFields()
+                    .IncludingProperties();
+
+                var typedSubject = (T)assertions.Subject;
```

Comment:
🤔 What if the subject is not of type `T`?
🤔 What if the subject is `null`?

Reply:
With respect to what to do if the subject is `null`: Both `BeBinarySerializable()` and `BeDataContractSerializable()` fail the assertion with what looks like a `NullReferenceException`. I could do the same, but I was thinking of something more like this:
```
    Execute.Assertion.ForCondition(assertions.Subject != null)
        .BecauseOf(because, becauseArgs)
        .FailWith("Expected {object:context} to be JSON serializable{reason}, but the value is null.  Please provide a value for the assertion.");
```
Not quite sure on the fail message, but it gets the point across I think

Reply:
Yeah, that's sloppy old code. The main library will do exactly what you suggested.

Reply:
Done :-)